openverse_catalog/dags/providers/provider_api_scripts/metropolitan_museum.py

```diff
@@ -6,138 +6,168 @@
+        other_images = object_json.get("additionalImages", [])
+        image_list = [main_image] + other_images
+
+        meta_data = self._get_meta_data(object_json)
+        raw_tags = self._get_tag_list(object_json)
+        title = self._get_title(object_json)
+        artist = self._get_artist_name(object_json)
+
+        # We aren't currently populating creator_url. In theory we could url encode
+        # f"https://collectionapi.metmuseum.org/public/collection/v1/search?artistOrCulture={artist}"
+        # per API guide here: https://metmuseum.github.io/#search
+        # but it seems fairly buggy (i.e. nonresponsive), at least when tested with
+        # "Chelsea Porcelain Manufactory" and "Minton(s)" and "Jean Pucelle"
+
+        return [
+            {
+                "foreign_landing_url": object_json.get("objectURL"),
+                "image_url": img,
+                "license_info": self.DEFAULT_LICENSE_INFO,
+                "foreign_identifier": self._get_foreign_id(object_id, img),
+                "creator": artist,
+                "title": title,
+                "meta_data": meta_data,
+                "filetype": img.split(".")[-1],
+                "raw_tags": raw_tags,
+            }
+            for img in image_list
+        ]
+
+    def _get_foreign_id(self, object_id: int, image_url: str):
+        unique_identifier = image_url.split("/")[-1].split(".")[0]
+        return f"{object_id}-{unique_identifier}"
+
+    def _get_meta_data(self, object_json):
+        meta_data = None
+        if object_json.get("accessionNumber") is not None:
+            meta_data = {
+                "accession_number": object_json.get("accessionNumber"),
+            }
+        return meta_data
+
+    def _get_tag_list(self, object_json):
+        tag_list = [
+            tag
+            for tag in [
+                object_json.get("department"),
+                object_json.get("medium"),
+                object_json.get("culture"),
+                object_json.get("objectName"),
+                self._get_artist_name(object_json),
+                object_json.get("classification"),
+                object_json.get("objectDate"),
+                object_json.get("creditLine"),
+                object_json.get("period"),
+            ]
+            if tag
+        ]
+        if object_json.get("tags"):
+            tag_list += [tag["term"] for tag in object_json.get("tags")]
+        return tag_list
+
+    def _get_title(self, record):
+        if record.get("title"):
+            return record.get("title")
+        else:
+            return record.get("objectName")
+
+    def _get_artist_name(self, record):
+        artist = record.get("artistDisplayName")
+        # Treating "unidentified" the same as missing, but maybe it would be useful in
+        # in search? Maybe in the art world it has a more specific outsider art
+        # connotation?
+        if artist in ["Unidentified", "Unidentified artist"]:
```

Comment:
Madison usually says that it's better to use a `set` when checking with `in` :)

Reply:
🤦  Yes, that makes total sense, I'll fix it tonight.

Reply:
Ha, you know me too well @obulat 😁

Reply:
But it actually does make a noticeable difference! ~7%!

```
(openverse) % python3 -m timeit '[n in [1, 2] for n in range(4)]'
1000000 loops, best of 5: 390 nsec per loop
(openverse) % python3 -m timeit '[n in {1, 2} for n in range(4)]'
1000000 loops, best of 5: 360 nsec per loop
```